File: sound-parent/sound-api/src/main/java/com/wisdom/sound/service/ResourceService.java
package com.wisdom.sound.service;
import java.util.HashMap;
import java.util.List;


import com.wisdom.sound.entity.PageResult;
import com.wisdom.sound.pojo.Resource;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ResourceService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Resource> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Resource resource);
	
	
	/**
	 * 修改
	 */
	public void update(Resource resource);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Resource findOne(String id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(String[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Resource resource, int pageNum, int pageSize);

	List<HashMap<String, String>> findForSelect2();
}

File: sound-parent/sound-web/src/main/resources/static/js/service/resourceService.js
//服务层
app.service('resourceService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/resource/findAll');
	}
	this.findForSelect2=function(){
		return $http.get('/resource/findForSelect2');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('/resource/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/resource/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/resource/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/resource/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/resource/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('/resource/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});
